### PR TITLE
add functional nn ops

### PR DESCRIPTION
**Pull request by justyx404**
So I paused at `.layernorm` and `.batchnorm` cause I feel like there's going to be overlaps between `BatchNorm2d`, `LayerNorm` that's from `nn`. So far my thinking is to make a link to the other md file (`__init__.py`) and just add examples there. Let me know what you think.

### CODE REVIEW

**Comment by wozeparrot** [comment]
I think it might be better to have the functional functions to have the examples and just put links in the `nn` documentation to the functional ones. Cuz the `nn` layers are just calling the functional ops anyways, just with some stored state.

**Comment by justyx404** [comment]
That works! Pytorch docs does the opposite but their structure is also a lot difference than ours.

**Comment by wozeparrot** [comment]
Yea, I think pytorch leans more layer first, while tinygrad leans more towards functional first.
